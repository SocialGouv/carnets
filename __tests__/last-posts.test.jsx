import { vi, expect, test } from "vitest";
import { render } from "@testing-library/react";
import Page from "../src/app/posts/page";

vi.mock("@/components/teams", () => {
  return {
    __esModule: true,
    default: () => {
      return <div />;
    },
  };
});

vi.mock("@/actions/get-last-posts", async (importOriginal) => {
  const actual = await importOriginal();

  return {
    ...actual,
    default: () => [
      {
        id: "1be7bfd2-ed4a-43df-b1be-b7502b544755",
        mood: "average",
        term: "",
        needs: "",
        author: "revolunet",
        team_slug: "sre",
        priorities:
          "## WIP\n" +
          "\n" +
          " - Migrations OVH en cours (46%) et progressivement dans les semaines à venir\n" +
          "\n" +
          "## Nouveaux services\n" +
          "\n" +
          " - BitWarden (gestion mots de passe équipes)\n" +
          " - Service d'antivirus pour les produits",
        created_at: "2023-12-14T10:50:24.261968+00:00",
        kpis: [
          {
            id: "d0bdf09f-c13e-4b3c-9712-0a6bb026bdaa",
            name: "Migration OVH",
            value: "46%",
          },
        ],
        team: {
          name: "SRE",
          avatarUrl:
            "https://avatars.githubusercontent.com/t/3729659?s=400&v=4",
        },
      },
    ],
  };
});

test("Page", async () => {
  const { container } = render(await Page());
  expect(container).toMatchSnapshot();
});
