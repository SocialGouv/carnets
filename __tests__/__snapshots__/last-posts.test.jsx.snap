// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page 1`] = `
<div>
  <aside
    class="flex-initial w-80"
  >
    <div />
  </aside>
  <main
    class="flex-1 pt-12 mb-8"
  >
    <section
      class="posts"
    >
      <button
        aria-controls="delete-post"
        aria-hidden="true"
        class="fr-btn fr-hidden"
        data-fr-opened="false"
        id="delete-post-hidden-control-button"
        tabindex="-1"
        type="button"
      >
         
      </button>
      <dialog
        aria-labelledby="fr-modal-title-delete-post"
        class="fr-modal"
        data-fr-concealing-backdrop="true"
        id="delete-post"
        role="dialog"
      >
        <div
          class="fr-container fr-container--fluid fr-container-md"
        >
          <div
            class="fr-grid-row fr-grid-row--center"
          >
            <div
              class="fr-col-12 fr-col-md-8 fr-col-lg-6"
            >
              <div
                class="fr-modal__body"
              >
                <div
                  class="fr-modal__header"
                >
                  <button
                    aria-controls="delete-post"
                    class="fr-btn--close fr-btn"
                    title="Fermer"
                    type="button"
                  >
                    Fermer
                  </button>
                </div>
                <div
                  class="fr-modal__content"
                >
                  <h1
                    class="fr-modal__title"
                    id="fr-modal-title-delete-post"
                  >
                    Suppression
                  </h1>
                  <div
                    class="relative"
                  >
                    <div
                      class=""
                    >
                      <p
                        class="my-8"
                      >
                        Êtes-vous sur de vouloir supprimer cette publication ?
                      </p>
                      <button
                        class="fr-btn fr-icon-delete-line fr-btn--icon-left"
                        id="fr-button-:r1:"
                        title="supprimer"
                      >
                        Supprimer
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </dialog>
      <article
        class="post"
      >
        <div
          class="header"
        >
          <div
            class="avatar"
          >
            <img
              alt="Picture of the team"
              data-nimg="1"
              decoding="async"
              height="48"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Ft%2F3729659%3Fs%3D400%26v%3D4&w=96&q=75"
              srcset="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Ft%2F3729659%3Fs%3D400%26v%3D4&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Ft%2F3729659%3Fs%3D400%26v%3D4&w=96&q=75 2x"
              style="color: transparent;"
              width="48"
            />
          </div>
          <div
            class="text"
          >
            SRE
          </div>
          <i
            class="flex items-center before:w-10 before:h-10 fr-icon-cloudy-2-line"
          />
        </div>
        <div
          class="kpis"
        >
          <div
            class="kpi"
          >
            <div
              class="value"
            >
              46%
            </div>
            <div
              class="name"
            >
              Migration OVH
            </div>
          </div>
        </div>
        <div
          class="tab-panel"
        >
          <div
            class="tabs"
          >
            <div
              class="tab selected"
            >
              <button
                class="fr-btn fr-btn--tertiary-no-outline"
                id="fr-button-:r2:"
              >
                Priorités
              </button>
            </div>
            <div
              class="tab"
            >
              <button
                class="fr-btn fr-btn--tertiary-no-outline"
                disabled=""
                id="fr-button-:r3:"
              >
                Besoins
              </button>
            </div>
            <div
              class="tab"
            >
              <button
                class="fr-btn fr-btn--tertiary-no-outline"
                disabled=""
                id="fr-button-:r4:"
              >
                Échéances
              </button>
            </div>
          </div>
          <div
            class="panels"
          >
            <div
              class="panel selected"
            >
              <div
                class="markdown-body"
              >
                <h2>
                  WIP
                </h2>
                

                <ul>
                  

                  <li>
                    Migrations OVH en cours (46%) et progressivement dans les semaines à venir
                  </li>
                  

                </ul>
                

                <h2>
                  Nouveaux services
                </h2>
                

                <ul>
                  

                  <li>
                    BitWarden (gestion mots de passe équipes)
                  </li>
                  

                  <li>
                    Service d'antivirus pour les produits
                  </li>
                  

                </ul>
              </div>
            </div>
            <div
              class="panel"
            >
              <div
                class="markdown-body"
              />
            </div>
            <div
              class="panel"
            >
              <div
                class="markdown-body"
              />
            </div>
          </div>
        </div>
        <div
          class="footer"
        >
          <div
            class="info"
          >
            14/12/2023
             - 
            <a
              class="mr-2"
              href="https://github.com/revolunet"
              rel="noreferrer"
              target="_blank"
            >
              revolunet
            </a>
          </div>
        </div>
      </article>
    </section>
  </main>
</div>
`;
